src/oemof/thermal/facades.py

```diff
@@ -0,0 +1,330 @@
+    for the operation is given as:
+    .. math::
+        x^{opex} = \sum_t (x^{flow, out}(t) \cdot c^{marginal\_cost}(t))
+
+    Examples
+    ---------
+    >>> from oemof import solph
+    >>> from oemof.thermal.facades import StratifiedThermalStorage
+    >>> heat_bus = solph.Bus(label='heat_bus')
+    >>> thermal_storage = StratifiedThermalStorage(
+    ...     label='thermal_storage',
+    ...     bus=heat_bus,
+    ...     diameter=10,
+    ...     height=10,
+    ...     temp_h=95,
+    ...     temp_c=60,
+    ...     temp_env=10,
+    ...     u_value=0.3,
+    ...     initial_storage_level=0.5,
+    ...     min_storage_level=0.05,
+    ...     max_storage_level=0.95
+    ...     capacity=1)
+    """
+
+    def __init__(self, *args, **kwargs):
+
+        super().__init__(
+            _facade_requires_=[
+                "bus", "diameter",
+                "temp_h", "temp_c", "temp_env",
+                "u_value"], *args, **kwargs
+        )
+
+        self.diameter = kwargs.get("diameter")
+
+        self.height = kwargs.get("height")
+
+        self.temp_h = kwargs.get("temp_h")
+
+        self.temp_c = kwargs.get("temp_c")
+
+        self.temp_env = kwargs.get("temp_env")
+
+        self.u_value = kwargs.get("u_value")
+
+        self.water_properties = {
+            'heat_capacity': kwargs.get("heat_capacity"), 'density': kwargs.get("density")
+        }
+
+        self.capacity = kwargs.get("capacity")
+
+        self.storage_capacity_cost = kwargs.get("storage_capacity_cost")
+
+        self.capacity_cost = kwargs.get("capacity_cost")
+
+        self.storage_capacity_potential = kwargs.get(
+            "storage_capacity_potential", float("+inf")
+        )
+
+        self.capacity_potential = kwargs.get(
+            "capacity_potential", float("+inf")
+        )
+
+        self.minimum_storage_capacity = kwargs.get(
+            "minimum_storage_capacity", 0
+        )
+
+        self.expandable = bool(kwargs.get("expandable", False))
+
+        if self.expandable and self.capacity is None:
+            self.capacity = 0
+
+        self.efficiency = kwargs.get("efficiency", 1)
+
+        self.marginal_cost = kwargs.get("marginal_cost", 0)
+
+        self.input_parameters = kwargs.get("input_parameters", {})
+
+        self.output_parameters = kwargs.get("output_parameters", {})
+
+        losses = calculate_losses(
+            self.u_value,
+            self.diameter,
+            self.temp_h,
+            self.temp_c,
+            self.temp_env,
+            **{key: value for key, value in self.water_properties.items() if value is not None}
+        )
+
+        self.loss_rate = losses[0]
+
+        self.fixed_losses_relative = losses[1]
+
+        self.fixed_losses_absolute = losses[2]
+
+        self.build_solph_components()
+
+    def build_solph_components(self):
+        """
+        """
+        self.inflow_conversion_factor = sequence(self.efficiency)
```

Comment:
```suggestion
        # self.inflow_conversion_factor = sequence(self.efficiency)
```

Just as with 'storage_capacity_cost', there is no consequence if this argument is missing.

Reply:
I guess this is because in GenericStorage, inflow_conversion_factor has a default value of 1.